src/net_processing.cpp

```diff
@@ -1283,22 +1286,57 @@ PeerManagerImpl::PeerManagerImpl(const CChainParams& chainparams, CConnman& conn
     // same probability that we have in the reject filter).
     m_recent_confirmed_transactions.reset(new CRollingBloomFilter(48000, 0.000001));
 
+    if (enable_rebroadcast) {
+        m_txrebroadcast = std::make_unique<TxRebroadcastHandler>(m_mempool, m_chainman, m_chainparams);
+
+        // Run fee rate cache every minute
+        scheduler.scheduleEvery([this] { m_txrebroadcast->CacheMinRebroadcastFee(); }, std::chrono::minutes{1});
```

Comment:
yup, here are some values are for running the whole function. Note that it does not distinguish CPU time vs time spent waiting to acquire the lock, as it currently acquires `cs_main` and `mempool.cs`, but the first might be unnecessary (see https://github.com/bitcoin/bitcoin/pull/21061#discussion_r602283867): 

the median value is 9090 μs, with 75% of the runs being under 11300 μs, and 99% of the run being under 11826 μs.

> Not sure if the block min feerate accuracy is worth the CPU time...

its a pretty critical component to this approach, without it the node would most likely be attempting to rebroadcast `MAX_REBROADCAST_WEIGHT` of transactions on every block.

Reply:
> the median value is 9090 μs, with 75% of the runs being under 11300 μs, and 99% of the run being under 11826 μs.

Okay and with a fulfilled mempool of the default size 300 MiB ? If yes I agree that's not that much and I think we're relieved on this concern :) If it does causes issues on low-grade nodes in the future, due to non-linear scaling of the CPU time in function of mempool size, I guess we'll be able to make this value configurable, offering a trade-off between CPU and bandwidth to node operators.

> its a pretty critical component to this approach, without it the node would most likely be attempting to rebroadcast MAX_REBROADCAST_WEIGHT of transactions on every block.

Do you mean without it or with a lower frequency rather than 1 min ? IIUC, the node would attempt to rebroadcast `MAX_REBROADCAST_WEIGHT` because you don't account for the fresh arrivals of new transactions. The feerate of those newcomers making most of your rebroadcast candidates irrelevant to block confirmation ? If refreshing the cache was costly, I guess we would run it once between _expected_ block intervals when the odds of having discovered enough new transactions to have a consistent view of next block are high. But cost is pretty low, so I don't think we care further... 

nit: maybe replace 1 by a named constant like `CACHE_REBROADCAST_FEERATE_FREQUENCY` ?

Reply:
> Okay and with a fulfilled mempool of the default size 300 MiB ? If yes I agree that's not that much and I think we're relieved on this concern :)

my mempool is bumped up to 1000MiB, so these are values for a significantly-larger-than-normal mempool :)

> I guess we'll be able to make this value configurable

yup, rebroadcast currently defaults off, but the plan is to keep this toggle in the long run to address any bandwidth concerns.

Reply:
> maybe replace 1 by a named constant like `CACHE_REBROADCAST_FEERATE_FREQUENCY`

done